toplevel/genprintval.ml

```diff
@@ -380,20 +381,31 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
               try
                 let decl = Env.find_type path env in
                 match decl with
-                | {type_kind = Type_abstract; type_manifest = None} ->
+                | {type_kind = Type_abstract _; type_manifest = None} ->
                     Oval_stuff "<abstr>"
-                | {type_kind = Type_abstract; type_manifest = Some body} ->
+                | {type_kind = Type_abstract _; type_manifest = Some body} ->
                     tree_of_val depth obj
                       (instantiate_type env decl.type_params ty_list body)
-                | {type_kind = Type_variant constr_list; type_unboxed} ->
-                    let unbx = type_unboxed.unboxed in
-                    let tag =
-                      if unbx then Cstr_unboxed
-                      else if O.is_block obj
-                      then Cstr_block(O.tag obj)
-                      else Cstr_constant(O.obj obj) in
+                | {type_kind = Type_variant (constr_list,rep)} ->
+                    let cstrs =
```

Comment:
I've added a comment.  This code got slightly more complicated because it used to be possible to tell from a `constructor_declaration` if the constructor is immediate, but now you need the `constructor_description` (at least, if you don't want to recompute layouts - because what used to be a check that there are no arguments is now a check that all the arguments are void).  It's possible we should put layout information into the declarations for convenience, though it doesn't quite feel like it belongs there.